src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -78,14 +77,15 @@ struct KeyValueViewModel: Comparable {
                 path: "client."
             )
         } else {
-            self.init(key: name, value: "?????")
+            self.init(key: param.name, value: "")
         }
     }
 
-    private init(param: ParameterType, constantSchema: ConstantSchema, name: String) {
+    private init(param: ParameterType, constantSchema: ConstantSchema) {
         self.optional = false
         self.needDecodingInMethod = false
         self.path = ""
+        let name = (param.paramLocation == .header) ? (param.serializedName ?? "") : param.name
```

Comment:
I'm wondering how much we can really special-case this. In any case, an assert would be appropriate here.

Reply:
The key in this case is the Http header value like Content-Type, Accept (all are upper cases.

Reply:
The assert would be for a blank key.

Reply:
Added an assert